### PR TITLE
WT-9899 Validate that bounded cursor logic doesn't apply to random cursors

### PR DESCRIPTION
**Pull request by clarissecheah**
Disable setting bounds with random cursors and add scenario in test_cursor_bound01 to validate this behaviour.

### CODE REVIEW

File: src/cursor/cur_std.c [hunk 1]
```diff
@@ -198,6 +198,7 @@ __wt_cursor_set_notsup(WT_CURSOR *cursor)
      * reset all of the cursors in a session. Reconfigure is left open in case it's possible in the
      * future to change these configurations.
      */
+    cursor->bound = __wt_cursor_config_notsup;
```
Comment (luke-pearson) [inline src/cursor/cur_std.c:201]: So the random cursor goes through this code path during its setup?

Comment (clarissecheah) [reply src/cursor/cur_std.c:201]: Yes, here's the relevant portion from cur_table.c:

```
WT_ERR(__wt_config_gets_def(session, cfg, "next_random", 0, &cval));
    if (cval.val != 0) {
        __wt_cursor_set_notsup(cursor);
        cursor->next = __curtable_next_random;
        cursor->reset = __curtable_reset;
    }
```

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**